### ISSUE TITLE
Timeout in communication after http request

### ISSUE THREAD

**Issue by Mafagafodotino** [miguelbalboa/rfid#407]
OS version: Windows 7
Arduino IDE version: 1.8.5
MFRC522 Library version: v1.4.0
ESP8266
MFRC522 device: RFID-RC522
  
I am having problems writing an information in a TAG MIFARE 1K, in my code I can authenticate by key a or be read a block of information without problems, but after that my code performs a POST and if the answer is as expected the reader should wait for a new reading to write the information that the request received in the TAG but when I try to write always results in timeout on the second authentication.

The information comes in a server response, the code starts when a card is detected, when this happens, I access a certain block of memory and verify by a "signature" that will guarantee that it is a valid card, if everything from right up to here the next step is to perform a POST for a script that searches for information in the database, in case the requested information exists, I must approach the TAG again from the reader, and with this, write in the TAG the information that the response of the script provide, and here begins the problem, when I try to write in the TAG this new authentication information fails.
I am using the mfrc522.PICC_HaltA () and mfrc522.PCD_StopCrypto1 () methods after getting the data contained in the TAG as I have seen being recommended in other doubts here and I also call the method that awakens the TAGs before looking for new TAG's in the reader.

After confirming that it is a TAG that does not yet have data, the code searches in a database for information to write in the TAG, if they exist, the reader must write them in the TAG but when I try to authenticate through the keys always results in timeout. I obtained a temporary success when I began to consider using KEY B to join instead of only KEY A to authenticate, read in the block where the signature is with KEY A, and to write in the TAG authenticated by KEY B, it worked in first TAG but in the second it returned to timeout in the communication, with the detail that both KEY A and KEY B are accepting the default code 0xFF ..

I already tried restarting the reader with PCD_Reset () and it did not work, the same goes for changing the status of the RST pin without result.

```c++
void init_Leitor () {
  SPI.begin ();
  delay (100);

  mfrc522.PCD_Init ();
}

void setup () {
  Serial.begin (115200);
  delay (10);
  conect_Wifi (ssid, wifi_password);
  init_Leitor ();

  // KEY FOR AUTHENTICATION IN TAG MEMORY SECTORS
  for (byte i = 0; i <6; i ++) {
    key1.keyByte [i] = 0xFF;
    yield ();
  }
}

void loop () {
  
  wake_RFID ();

  Serial.print (F ("\ r \ nFollow a card in the reader ... \ r \ n"));
  while (! mfrc522.PICC_IsNewCardPresent () &&! mfrc522.PICC_ReadCardSerial ()) {
    yield ();
  }

  auth_A ();

  reading_TAG ();

  close_RFID ();
}

void wake_RFID () {
  byte bufferATQA [2];
  byte bufferSize = sizeof (bufferATQA);
  mfrc522.PICC_WakeupA (bufferATQA, & bufferSize);
}

void auth_A () {
  dump_byte_array (mfrc522.uid.uidByte, mfrc522.uid.size);

  status = (MFRC522 :: StatusCode) mfrc522.PCD_Authenticate (MFRC522 :: PICC_CMD_MF_AUTH_KEY_A, trailer_Block, & key1, & (mfrc522.uid));
  if (status! = MFRC522 :: STATUS_OK) {
    Serial.print (F ("PCD_Authenticate () failed:"));
    Serial.println (mfrc522.GetStatusCodeName (status));
    return;
  }
 
  Serial.println (F ("Data currently in the sector:"));
  mfrc522.PICC_DumpMifareClassicSectorToSerial (& (mfrc522.uid), & key1, sector_Memory);
  Serial.println ();
}

void written_TAG () {
  // WRITTEN
  Serial.print ("Writing data in block" + (String) blockAddr + "\ r \ n");
  dump_byte_array (Info_block, 16); Serial.println ();
  byte aux1 [16];
  saldo.getBytes (aux1, 16);
  status = (MFRC522 :: StatusCode) mfrc522.MIFARE_Write (byte (ad_addr), aux1, 16);
  if (status! = MFRC522 :: STATUS_OK) {
    Serial.print (F ("MIFARE_Write () failed:"));
    Serial.println (mfrc522.GetStatusCodeName (status));
  }
}

void read_TAG () {
  //READING
  Serial.print ("Reading block data" + (String) blockAddr + "\ r \ n");
  status = (MFRC522 :: StatusCode) mfrc522.MIFARE_Read (addr, buffer, & size);
  if (status! = MFRC522 :: STATUS_OK) {
    Serial.print (F ("\\ MIFARE_Read () failed:"));
    Serial.println (mfrc522.GetStatusCodeName (status));
  }
  // SHOW THE DATA CONTAINED ON THE SPECIFIED BLOCK
  Serial.print ("Data in block" + (String) blockAddr + ":");
  flag1 = true;
  dump_byte_array (buffer, 16);
  if (array_char [0] == 'S') {
    Serial.println ("Card already has data!");
    return;
  }
  else {
    Serial.println ("A new card is requesting data!");
    close_RFID ();
    // SEND POST REQUEST 1
  }
}

void close_RFID () {
  // END THE AUTHENTICATED SESSION BY KEY A
  mfrc522.PICC_HaltA ();
  // FOR READER PROCESSING
  mfrc522.PCD_StopCrypto1 ();
  delay (500);
}
```

This is the function where I read the responses of the scripts to my POST requests, I believe that the problem is here, if the answer is the expected one (the information searched for exists in the database), I start the reader again, but when I try to authenticate again (the reader is stopped before starting the functions as web-client) results in timeout.

```c++
void reply () {
  String response_line;
  Serial.println (F ("\ r \ nVerifying server response ... \ r \ n")));
  while (client.available ()> 0) {
    line_respon = client.readStringUntil ('\ r');
    Serial.print (response_line);
    if (input_response_indexOf ("Data exists")! = -1) {
      
      wake_RFID ();
      Serial.println (F ("\ r \ nConfirm receipt \ r \ n"));
      while (! mfrc522.PICC_IsNewCardPresent () &&! mfrc522.PICC_ReadCardSerial ()) {
        yield ();
      }
      auth_A ();
      writing_TAG ();
      close_RFID ();
      send_req2 (url_2, host);
    }
    if (answer_index.Of ("OK")! = -1) {
      Serial.print (F ("Confirmed \ r \ n"));
    }
  }
  end_http ();
}
```
Thanks in advance!

edit: next time please use syntax highlighting
